internal/controller/variantautoscaling_controller.go

```diff
@@ -508,27 +508,95 @@ func (r *VariantAutoscalingReconciler) readAcceleratorConfig(ctx context.Context
 	return out, nil
 }
 
-func (r *VariantAutoscalingReconciler) getPrometheusConfig(ctx context.Context) (string, error) {
-	// First, try environment variable
-	if promAddr := os.Getenv("PROMETHEUS_BASE_URL"); promAddr != "" {
-		logger.Log.Info("Using Prometheus address from environment variable -> ", "address: ", promAddr)
-		return promAddr, nil
+func (r *VariantAutoscalingReconciler) getPrometheusConfig(ctx context.Context) (*interfaces.PrometheusConfig, error) {
+	// Try environment variables first
+	if config, err := r.getPrometheusConfigFromEnv(); err != nil {
+		return nil, fmt.Errorf("failed to get config from environment: %w", err)
+	} else if config != nil {
+		logger.Log.Info("Using Prometheus configuration from environment variables")
+		return config, nil
 	}
 
-	// Then, try to get from ConfigMap with retry logic
+	// Try ConfigMap second
+	if config, err := r.getPrometheusConfigFromConfigMap(ctx); err != nil {
+		return nil, fmt.Errorf("failed to get config from ConfigMap: %w", err)
+	} else if config != nil {
+		logger.Log.Info("Using Prometheus configuration from ConfigMap")
+		return config, nil
+	}
+
+	// No configuration found
+	logger.Log.Warn("No Prometheus configuration found. Please set PROMETHEUS_BASE_URL environment variable or configure via ConfigMap")
+	return nil, fmt.Errorf("no Prometheus configuration found. Please set PROMETHEUS_BASE_URL environment variable or configure via ConfigMap")
+}
+
+func (r *VariantAutoscalingReconciler) getPrometheusConfigFromEnv() (*interfaces.PrometheusConfig, error) {
+	promAddr := os.Getenv("PROMETHEUS_BASE_URL")
+	if promAddr == "" {
+		return nil, nil // No config found, but not an error
+	}
+
+	logger.Log.Info("Using Prometheus address from environment variable -> ", "address: ", promAddr)
+	return utils.ParsePrometheusConfigFromEnv(), nil
+}
+
+func (r *VariantAutoscalingReconciler) getPrometheusConfigFromConfigMap(ctx context.Context) (*interfaces.PrometheusConfig, error) {
 	cm := corev1.ConfigMap{}
 	err := utils.GetConfigMapWithBackoff(ctx, r.Client, configMapName, configMapNamespace, &cm)
 	if err != nil {
-		logger.Log.Warn("Failed to get ConfigMap for Prometheus config after retries, using default - ", "error: ", err)
-	} else if promAddr, exists := cm.Data["PROMETHEUS_BASE_URL"]; exists && promAddr != "" {
-		logger.Log.Info("Using Prometheus address from ConfigMap -> ", "address: ", promAddr)
-		return promAddr, nil
+		return nil, fmt.Errorf("failed to get ConfigMap for Prometheus config: %w", err)
+	}
+
+	promAddr, exists := cm.Data["PROMETHEUS_BASE_URL"]
+	if !exists || promAddr == "" {
+		return nil, nil // No config found, but not an error
```

Comment:
Why is this not error, i thought we will only read prom URL from env var.

Reply:
This is the design:
Environment variable missing returns no error
ConfigMap missing returns no error
Both missing returns error after checking both